INN/umbrella-citylimits
# Allow article sidebar to be sticky

## Changes

This pull request makes the following changes:

- Wraps the single two column template content and sidebar in a parent div named `content-container`, which makes sure the sidebar is the same height as the content so that `position: sticky;` works on the sidebar.
- Updates `get_the_ID` to `get_the_ID()` in the `content-single.php` partial so there's no ugly `undefined constant` error

Note that we should probably warn them that they'll need/want to remove some widgets from the sidebar now. `position:sticky;` doesn't allow you to scroll in the sidebar so you either wait until you're scrolled 100% down the page, or you make the sidebar height less than the viewport height by reducing the number of widgets

Currently with a ton of widgets:
![ezgif-7-b6f2b755989c](https://user-images.githubusercontent.com/18353636/74263548-0f7bce00-4ccd-11ea-9057-e970808379bd.gif)

With only 2 widgets:
![ezgif-7-a6ff435a5b77](https://user-images.githubusercontent.com/18353636/74263542-0be84700-4ccd-11ea-9875-b8290519fa83.gif)

## Why



For #52

## Testing/Questions

Features that this PR affects:

- Single two column template and sidebar display


Questions that need to be answered before merging:

- [ ] Is anything broken in here?
- [ ] Is it possible to add `overflow: scroll;` to the sidebar? Every way that I have tried has broken the sticky positioning

Steps to test this PR:

1. View a post with a sidebar
2. Scroll, scroll, scroll and make sure the sidebar stays sticky!
